controllers/prep_handlers.go

```diff
@@ -327,28 +300,163 @@ func (r *ImageBasedUpgradeReconciler) SetupStateroot(ctx context.Context, ibu *l
 }
 
 func (r *ImageBasedUpgradeReconciler) launchSetupStateroot(ctx context.Context,
-	ibu *lcav1alpha1.ImageBasedUpgrade, progressfile string) (ctrl.Result, error) {
-	result := requeueImmediately()
-	if err := updateProgressFile(progressfile, "started-stateroot"); err != nil {
-		r.Log.Error(err, "failed to update progress file")
-		return result, err
-	}
-	if err := r.SetupStateroot(ctx, ibu); err != nil {
+	ibu *lcav1alpha1.ImageBasedUpgrade) (success bool, err error) {
+	if err = r.SetupStateroot(ctx, ibu); err != nil {
 		r.Log.Error(err, "failed to setup stateroot")
-		if err := updateProgressFile(progressfile, "Failed"); err != nil {
-			r.Log.Error(err, "failed to update progress file")
-			return result, err
+		success = false
+	} else {
+		success = true
+	}
+	return
+}
+
+func (r *ImageBasedUpgradeReconciler) setPrepFailedStatus(ibu *lcav1alpha1.ImageBasedUpgrade) {
+	utils.SetStatusCondition(&ibu.Status.Conditions,
+		utils.GetCompletedConditionType(lcav1alpha1.Stages.Prep),
+		utils.ConditionReasons.Completed,
+		metav1.ConditionFalse,
+		"Prep failed",
+		ibu.Generation)
+	utils.SetStatusCondition(&ibu.Status.Conditions,
+		utils.GetInProgressConditionType(lcav1alpha1.Stages.Prep),
+		utils.ConditionReasons.Completed,
+		metav1.ConditionFalse,
+		"Prep failed",
+		ibu.Generation)
+}
+
+func (r *ImageBasedUpgradeReconciler) setPrepSucceededStatus(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) {
```

Comment:
this function does not need to be part of reconciler. We can pass in the message.